File: Latihan2_2/cpp/Product.cpp
#include <iostream>
#include <string>
using namespace std;

// class Product adalah parent dari class Hardware
class Product{
    // atribut private
    private:
        string price;
        string idProduct;
    
    public:
        //constructor  kosong     
        Product(string price, string idProduct){
            this->price = price;
            this->idProduct = idProduct;
        }

        //constructor  kosong     
        Product(){
        }
        
        // set memory price
        void set_price(string price){
            this->price = price;
        }
        // get memory price
        string get_price(){
            return this->price;
        }

        // set memory idProduct
        void set_idProduct(string idProduct){
            this->idProduct = idProduct;
        }
        // get memory product
        string get_idProduct(){
            return this->idProduct;
        }

        // destructor
        ~Product(){
        }
};

File: Latihan2_2/python/Product.py
# class Product adalah parent dari class Hardware
class Product:
    # atribut private
    __price = ""
    __idProduct = ""

    # constructor
    def __init__(self):
        self.__price = ""
        self.__idProduct = ""
    
    # set product price
    def set_price(self, price):
        self.__price = price
    # get product price
    def get_price(self):
        return self.__price

    # set product price
    def set_idProduct(self, idProduct):
        self.__idProduct = idProduct
    # get product price
    def get_idProduct(self):
        return self.__idProduct